src/lab14.py

```diff
@@ -1153,20 +1215,25 @@ def render(self):
         if self.needs_layout:
             self.document = DocumentLayout(self.nodes)
             self.document.layout(self.zoom)
-            if self.accessibility_is_on:
-                self.needs_accessibility = True
-            else:
-                self.needs_paint = True
+            self.needs_accessibility = True
+            self.needs_paint = True
             self.needs_layout = False
 
         if self.needs_accessibility:
+            old_afocus = self.accessibility_focus
```

Comment:
could accessibility_node be an actual Node? And we dynamically compute which a11y node it refers to?

Reply:
We could, definitely, but that raises the question of why we need the accessibility tree at all. As far as I can tell, this is a constraint imposed by operating systems: their accessibility technologies rquire you to provide an accessibility tree...

Reply:
Sorry I was unclear. I meant to cache a Node, and then whenever code currently writes `self.accessibility_focus` we would instead write `self.accessibility_node.layout_object.accessiblity_object`, via tree forward pointers.

Reply:
Operating systems do not actually require there be a tree. They do require that the browser respond to requests synchronously though, and this requires them to have all the necessary parts of the DOM ready at hand. In a multi-process browser, this requires either serializing the DOM to another process, or some subset of it (the accessibility tree). It's a tree only because that is the reasonable and obvious choice for representing some subset of DOM stuff.